infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java

```diff
@@ -42,12 +55,21 @@ public KubernetesNamespaceFactory(
       @Nullable @Named("che.infra.kubernetes.cluster_role_name") String clusterRoleName,
       KubernetesClientFactory clientFactory) {
     this.namespaceName = namespaceName;
-    this.isPredefined = !isNullOrEmpty(namespaceName);
+    this.isPredefined = !isNullOrEmpty(namespaceName) && !hasPlaceholders(namespaceName);
     this.serviceAccountName = serviceAccountName;
     this.clusterRoleName = clusterRoleName;
     this.clientFactory = clientFactory;
   }
 
+  private boolean hasPlaceholders(String namespaceName) {
+    for (String placeholder : NAMESPACE_NAME_PLACEHOLDERS.keySet()) {
```

Comment:
wdyt may be stream-based implementation would look clener?

Reply:
might be, I don't have strong preference here. I've changed it to stream, but kept it in own method.